### PR TITLE
legacy-support: update to v1.2.4

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
- stpncpy: Fix potential header collision with "later" SDK
- provide workaround for pthread_get_stacksize_np

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by fhgwright** [comment]
@MarcusCalhoun-Lopez @mascguy 
I'll test this shortly.

**Comment by fhgwright** [comment]
TL; DR; Good to go, except possibly for @mascguy weighing in.

I tested this on the usual platforms:
```
Mac OS X 10.4.11 8S165, PPC, Xcode 2.5 8M2558
Mac OS X 10.4.11 8S2167, i386, Xcode 2.5 8M2558
Mac OS X 10.4.11 8S2167, x86_64, Xcode 2.5 8M2558
Mac OS X 10.5.8 9L31a, PPC, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, i386, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, x86_64, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, PPC (i386 Rosetta), Xcode 3.1.4 9M2809
Mac OS X 10.6.8 10K549, i386, Xcode 3.2.6 10M2518
Mac OS X 10.6.8 10K549, x86_64, Xcode 3.2.6 10M2518
Mac OS X 10.6.8 10K549, PPC (i386 Rosetta), Xcode 3.2.6 10M2518
Mac OS X 10.7.5 11G63, x86_64, Xcode 4.6.3 4H1503
OS X 10.8.5 12F2560, x86_64, Xcode 5.1.1 5B1008
OS X 10.9.5 13F1911, x86_64, Xcode 6.2 6C131e
OS X 10.10.5 14F2511, x86_64, Xcode 7.2 7C68
OS X 10.11.6 15G22010, x86_64, Xcode 8.1 8B62
macOS 10.12.6 16G2136, x86_64, Xcode 9.2 9C40b
macOS 10.13.6 17G14042, x86_64, Xcode 10.1 10B61
macOS 10.14.6 18G9323, x86_64, Xcode 11.3.1 11C505
macOS 10.15.7 19H15, x86_64, Xcode 12.4 12D4e
macOS 11.7.10 20G1427, x86_64, Xcode 13.2.1 13C100
macOS 11.7.10 20G1427, arm64, Xcode 13.2.1 13C100
macOS 12.7.4 21H1123, x86_64, Xcode 14.2 14C18
macOS 12.7.4 21H1123, arm64, Xcode 14.2 14C18
macOS 13.6.6 22G630, arm64, Xcode 15.2 15C500b
macOS 14.4.1 23E224, arm64, Xcode 15.3 15E204a
```
Aside from the usual build failure on 10.4 ppc +universal, it builds and passes tests on all of the above.  No surprise, since this is just a different presentation of previously tested code.

Feel free to paste this list into the "Tested on" section above.  I don't have edit rights to do that.

All of the items under "Verified" can be checked, except the ticket URL item, which is "N/A".

We should probably allow some time for @mascguy to weigh in, though he seems to be busy with other things right now.  I don't think it's necessary to wait the full 72-hour maintainer timeout, for a couple of reasons:

1. I'm already approving this as a maintainer, after fairly thorough testing.
2. @mascguy already approved this same code in its prerelease state.

I suggest waiting until this PR is at least 24 hours old, then merging it if @mascguy hasn't objected.

**Review by mascguy — APPROVED** [review]
Given how much work and testing you folks have put into this, let's merge.

Great collaboration, as always!

**Comment by mascguy** [comment]
Just queued rebuilds for Rust/Cargo, on the buildbots for 10.9 and 10.10.

**Comment by mascguy** [comment]
> Just queued rebuilds for Rust/Cargo, on the buildbots for 10.9 and 10.10.

Fortunately the buildbots were idle, and the builds for Rust are in-progress:

- https://build.macports.org/builders/ports-10.9_x86_64-builder/builds/279035
- https://build.macports.org/builders/ports-10.10_x86_64-builder/builds/271819

**Comment by fhgwright** [comment]
@mascguy 
> Just queued rebuilds for Rust/Cargo, on the buildbots for 10.9 and 10.10.

Did enough time elapse for the new `legacy-support` to appear on the ports tree mirrors, or do the buildbots bypass that?  The typically quoted delay is an hour, though in my experience it tends to be more like 1.5 hours, and I've at least once seen it take around 12 hours.

Once it's available, it would also make sense to queue a rebuild for `R-nloptr` on the 10.6 buildbot(s).

**Comment by mascguy** [comment]
> Did enough time elapse for the new `legacy-support` to appear on the ports tree mirrors, or do the buildbots bypass that?

The buildbots pull directly from Git master, so there's no delay.

> Once it's available, it would also make sense to queue a rebuild for `R-nloptr` on the 10.6 buildbot(s).

Ah, thanks for the heads-up, just queued that as well.

**Comment by fhgwright** [comment]
> > Once it's available, it would also make sense to queue a rebuild for `R-nloptr` on the 10.6 buildbot(s).
> 
> Ah, thanks for the heads-up, just queued that as well.

Well, after all the arguing about the `R-nloptr` fix, it would be nice to have it fully effective. :-)

**Comment by mascguy** [comment]
> Fortunately the buildbots were idle, and the builds for Rust are in-progress:
> 
> - https://build.macports.org/builders/ports-10.9_x86_64-builder/builds/279035
> - https://build.macports.org/builders/ports-10.10_x86_64-builder/builds/271819

Marcus, it looks like the bootstrap build succeeds. But both fail to find a macOS SDK during the LLVM build phase. (Just wanted to provide a quick heads-up; we can discuss further via Trac, or wherever you prefer.)

**Comment by MarcusCalhoun-Lopez** [comment]
> But both fail to find a macOS SDK during the LLVM build phase. (Just wanted to provide a quick heads-up; we can discuss further via Trac, or wherever you prefer.)

Unbeknownst to me, it seems that early command line tools did not install SDKs.
Hopefully, https://github.com/macports/macports-ports/pull/24125 fixes the problem.
